sound/soc/sof/trace.c

```diff
@@ -13,6 +13,55 @@
 #include "sof-priv.h"
 #include "ops.h"
 
+static ssize_t sof_dfsentry_trace_filter_write(struct file *file,
+		const char __user *from, size_t count, loff_t *ppos)
+{
+	struct snd_sof_dfsentry *dfse = file->private_data;
+	struct snd_sof_dev *sdev = dfse->sdev;
+	char *string;
+	int ret;
+
+	string = kmalloc(count, GFP_KERNEL);
+	if (!string) {
+		ret = -ENOMEM;
+		goto error;
+	}
+
+	ret = simple_write_to_buffer(string, count, ppos, from, count);
+	if (ret != count)
+		ret = ret >= 0 ? -EIO : ret;
+
+error:
+	if (string)
+		kfree(string);
+	return ret;
```

Comment:
Commits order can't be reversed, because then I introduce unused static functions. I split changes into two commits, to make them shorter and to distinguish part related more with kernel system usage from functional code.